apps/api/src/app/workflows-v2/usecases/build-step-issues/build-step-issues.usecase.ts

```diff
@@ -304,4 +311,36 @@ export class BuildStepIssuesUsecase {
 
     return issues.controls?.skip.length ? issues : {};
   }
+
+  @Instrument()
+  private async validateIntegration(args: {
+    stepTypeDto: StepTypeEnum;
+    environmentId: string;
+    organizationId: string;
+  }): Promise<StepIssuesDto> {
+    const issues: StepIssuesDto = {};
+
+    const validIntegrationForStep = await this.integrationsRepository.findOne({
+      _environmentId: args.environmentId,
+      _organizationId: args.organizationId,
+      active: true,
```

Comment:
Having the `active` flag is enough for the `push, in-app, chat` channels, but for the `email, sms` we also have the `primary` which basically tells whether the integration will be used during sending phase.

So what I'm saying is that even if the integration is `active` it doesn't mean that we will send the notification for the `email, sms` channels. Users should be informed that they don't have a primary integration set in that case.

So we should have two types of issues for the integrations.

@desiprisg @SokratisVidros

Reply:
@LetItRock Are you suggesting that the check is `{ active: true, primary: true }` and treat them as one error?

Reply:
I have added an extra check in case of email or sms and adjusted the issue msg.